ui/lib/pki/addon/components/pki-issuer-cross-sign.js

```diff
@@ -0,0 +1,145 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import errorMessage from 'vault/utils/error-message';
+import { waitFor } from '@ember/test-waiters';
+/**
+ * @module PkiIssuerCrossSign
+ * PkiIssuerCrossSign components render from a parent issuer's details page to cross-sign an intermediate issuer.
+ * The component reads an existing intermediate issuer, cross-signs it with a parent issuer and imports the new
+ * issuer into an existing intermediate mount using three inputs from the user:
+ * intermediateMount (the mount where the issuer to be cross signed lives)
+ * intermediateIssuer (the name of the intermediate issuer, located in the above mount)
+ * newCrossSignedIssuer (the name of the to-be-cross-signed intermediate issuer)
+ *
+ * The requests involved and how those inputs are used:
+ * 1. Read an existing intermediate issuer
+ *    -> GET /:intermediateMount/issuer/:intermediateIssuer
+ * 2. Create a new CSR based on this existing issuer ID
+ *    -> POST /:intermediateMount/intermediate/generate/existing
+ * 3. Sign it with the new parent issuer, minting a new certificate.
+ *    -> POST /this.args.parentIssuer.backend/issuer/this.args.parentIssuer.issuerName/sign-intermediate
+ * 4. Import it back into the existing mount
+ *    -> POST /:intermediateMount/issuers/import/bundle
+ * 5. Read the imported issuer
+ *    -> GET /:intermediateMount/issuer/:issuer_id
+ * 5. Update this issuer with the newCrossSignedIssuer
+ *    -> POST /:intermediateMount/issuer/:issuer_id
```

Comment:
Great comments! I wonder if these would be better in the issuer adapter though?

Reply:
These aren't specific to the `pki/issuer` since cross-signing bounces between from`pki/issuer` to `pki/action` to `pki/issuer` 

I'm happy to move them elsewhere! I just figured that a gist of the order of operations happening in this component might be helpful for future debugging